src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Vault/SetAzureSiteRecoveryVaultSettings.cs

```diff
@@ -29,6 +29,13 @@ public class SetAzureSiteRecoveryVaultSettings : SiteRecoveryCmdletBase
         #region Parameters
 
         /// <summary>
+        /// Gets or sets vault Object.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.Default, Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public System.Object Vault { get; set; }
```

Comment:
The problem is, how will the user know how to create this type - there is no indication of what properties it should have and its value cannot come from the pipeline, because you have attributes it to exclude this.  Please change this parameter type to be a strongly typed parameter.  In the worst case, you can have two parametsr sets, one that accepts ASRVault and the other that acepts ARSVault.  As it is, this cmdlet will be very difficult to use.